ProcessOptimizer/samplers/doe/optimal_design.py

```diff
@@ -0,0 +1,697 @@
+import math
+
+import numpy as np
+import patsy
+import scipy
+
+from ProcessOptimizer.space import Categorical
+from .doe_transform import doe_to_real_space
+from .doe_utils import (generate_replicas_and_sort, round_design_point_values,
+                        sanitize_names_for_patsy)
+
+# Defining a number of helper functions for the optimal design of experiments
+
+
+def hit_and_run(x0, constraint_matrix, bounds, n_samples, thin=1, seed=None):
+    """A basic implementation of the hit and run sampler
+
+    :param x0: The starting value of sampler.
+    :param constraint_matrix: A matrix of constraints in the form Ax <= b.
+    :param bounds: A vector of bounds in the form Ax <= b.
+    :param n_samples: The numbers of samples to return.
+    :param thin: The thinning factor. Retain every 'thin' sample
+        (e.g. if thin=2, retain every 2nd sample)
+    :param seed: possibility to specify a seed for random generator
+
+    This function is adapted from https://github.com/statease/dexpy
+    Copyright 2016 Stat-Ease, Inc.
+    License: Apache License, Version 2.0
+    License link: https://github.com/statease/dexpy/blob/master/LICENSE
+    """
+    x = np.copy(x0)
+    p = len(x)
+
+    if seed:
+        rng = np.random.RandomState(seed)
```

Comment:
Also, seed=0 will give random points, not points seeded with 0, since 0 is falsy